src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -295,6 +296,21 @@ private static PackageSpec ToPackageSpec(ProjectNames projectNames, IVsProjectRe
             return packageSpec;
         }
 
+        /// <summary>
+        /// Tries to get  the projectId for a project from the project system cache using the project unique name.
+        /// </summary>
+        /// <param name="projectUniqueName">Unique name for the project.</param>
+        /// <param name="cache">ProjectSystem cache</param>
+        /// <param name="result">Contains the projectId if the return result was true else it is set to null.</param>
+        /// <returns>Returns bool indicating if the operation was successful.</returns>
+        private static bool TryGetProjectIdFromCache(string projectUniqueName, IProjectSystemCache cache, out string result)
+        {
+            result = null;
```

Comment:
why do you need to set this to null here?

Reply:
out vars need to be set to null.